src/main/java/de/theshadowsdust/ultimatespawn/listener/SpawnLocationListener.java

```diff
@@ -24,12 +26,15 @@ public SpawnLocationListener(@NotNull UltimateSpawnPlugin plugin) {
 
     @EventHandler(priority = EventPriority.HIGHEST)
     public void handlePlayerSpawnLocation(PlayerSpawnLocationEvent event) {
+        Player player = event.getPlayer();
         try {
-            if (this.plugin.getConfigurationService().getConfig().isJoinAtSpawn()) {
-                SpawnPosition spawnPosition = this.plugin.getSpawnPositionService().getDefaultSpawn().get();
-                Location location = WrappedLocation.toLocation(spawnPosition.getWrappedLocation());
-                if (location == null) return;
-                event.setSpawnLocation(location);
+            if(!player.hasPlayedBefore()) {
+                event.setSpawnLocation(player.getWorld().getSpawnLocation());
+            }
```

Comment:
Add new config option to disable this behavior

Reply:
done, I guess. Can you look at this again @TheMeinerLP ?